File: galgamecommon/src/main/scala/com/goexp/galgame/common/model/DateRange.scala
package com.goexp.galgame.common.model

import java.time.LocalDate
import java.util.StringJoiner

class DateRange(val start: LocalDate, val end: LocalDate) {
  override def toString: String = new StringJoiner(", ", classOf[DateRange].getSimpleName + "[", "]").add("start=" + start).add("end=" + end).toString
}

File: galgamecommon/src/main/scala/com/goexp/galgame/common/model/DateRange.java
package com.goexp.galgame.common.model;

import java.time.LocalDate;
import java.util.StringJoiner;

public class DateRange {

    public final LocalDate start;

    public final LocalDate end;

    public DateRange(LocalDate start, LocalDate end) {
        this.start = start;
        this.end = end;
    }

    @Override
    public String toString() {
        return new StringJoiner(", ", DateRange.class.getSimpleName() + "[", "]")
                .add("start=" + start)
                .add("end=" + end)
                .toString();
    }
}
